rust/otap-dataflow/crates/otap/src/syslog_cef_receiver.rs

```diff
@@ -964,3 +1126,188 @@ mod tests {
+
+            // Receiver with metrics enabled via pipeline
+            let receiver = SyslogCefReceiver::with_pipeline(
+                pipeline,
+                Config::new(listening_addr, Protocol::Udp),
+            );
+
+            // Keep downstream open to avoid refused
+            let (out_tx, mut _out_rx) = otap_df_channel::mpsc::Channel::new(8);
+            let mut senders = std::collections::HashMap::new();
+            let _ = senders.insert(
+                "".into(),
+                otap_df_engine::local::message::LocalSender::MpscSender(out_tx),
+            );
+
+            let (pipe_tx, _pipe_rx) = otap_df_engine::control::pipeline_ctrl_msg_channel(10);
+            // Telemetry reporter for effect handler
+            let (metrics_rx, reporter) = MetricsReporter::create_new_and_receiver(4);
+            let eh = otap_df_engine::local::receiver::EffectHandler::new(
+                test_node("syslog_udp_ok"),
+                senders,
+                None,
+                pipe_tx,
+                reporter.clone(),
+            );
+
+            let (ctrl_tx, ctrl_rx) = otap_df_channel::mpsc::Channel::new(16);
+            let ctrl_rx = otap_df_engine::message::Receiver::Local(
+                otap_df_engine::local::message::LocalReceiver::MpscReceiver(ctrl_rx),
+            );
+            let ctrl_chan = otap_df_engine::local::receiver::ControlChannel::new(ctrl_rx);
+
+            // Start receiver
+            let handle = tokio::task::spawn_local(async move {
+                let _ = Box::new(receiver).start(ctrl_chan, eh).await;
+            });
+
+            // Send one valid and one invalid UDP datagram
+            let sock = UdpSocket::bind("127.0.0.1:0").await.unwrap();
+            let _ = sock
+                .send_to(
+                    b"<34>1 2024-01-15T10:30:45.123Z host app - ID1 msg",
+                    listening_addr,
+                )
+                .await
+                .unwrap();
+            // Our RFC3164 parser accepts arbitrary non-empty strings as content-only messages.
+            // To exercise the "invalid" path, send an empty datagram which is rejected by the parser.
+            let _ = sock.send_to(b"", listening_addr).await.unwrap();
+
+            // Allow interval to tick
+            tokio::time::sleep(Duration::from_millis(150)).await;
+
+            // Trigger telemetry collection
+            let _ = ctrl_tx.send(NodeControlMsg::CollectTelemetry {
+                metrics_reporter: reporter.clone(),
+            });
+
+            // Shutdown
+            let _ = ctrl_tx.send(NodeControlMsg::Shutdown {
+                deadline: Instant::now(),
+                reason: "test".into(),
+            });
+            let _ = handle.await;
+
+            // Validate
+            let snapshot = metrics_rx.recv_async().await.unwrap();
+            let m = snapshot.get_metrics();
+            // Order: forwarded, invalid, forward_failed, total, tcp_connections_active
```

Comment:
Make assertions for all of them. For example, in this case, `forward_failed` and `tcp_connections_active` should be `0`.